### PR TITLE
Use a sync.Map for span storage

### PR DESCRIPTION
**Pull request by sdboyer-stripe**
This may eventually help with ingest throughput, though it makes no apparent difference when we're ingesting one span per RPC call, as that's a far narrower bottleneck than the mutex lock/unlock around the map that this eliminates.

A natural next step here (possibly for this PR) would be to eliminate standalone workers and just have everything run against a single `sync.Map` that's stored on the server object. (Unless i've missed something else, the only real point of the workers was an attempt at scaling, but now that the core datastructure is lock-free, there's no need for that).

### CODE REVIEW

File: generate.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
 package falconer
 
 //go:generate gojson -input example.yaml -o config.go -fmt yaml -pkg falconer -name Config
-//go:generate protoc --proto_path=vendor:. falconer.proto --go_out=plugins=grpc:.
```
Comment (ChimeraCoder) [inline generate.go]: What was the difference here? (I don't remember offhand, and I could look it up, but we should document it in the PR comments anyway.)

**Comment by ChimeraCoder** [comment]
one question - lgtm besides that

File: generate.go (hunk 1 above)
Comment (sdboyer-stripe) [reply generate.go]: It's just that the `--proto_path` (which is the same as `-I`) doesn't seem to take multiple colon-separated options sanely - that's all.

**Pull request merged**